components/nodemanager-service/pgdb/project_update.go

```diff
@@ -0,0 +1,267 @@
+	}
+	return nodesDaos, err
+}
+
+func getMatchingProjectIDs(node *NodeProjectData, projectRules map[string]*iam_v2.ProjectRules) []string {
+	matchingProjects := make([]string, 0)
+
+	for projectName, project := range projectRules {
+		if nodeMatchesRules(node, project.Rules) {
+			matchingProjects = append(matchingProjects, projectName)
+		}
+	}
+
+	return matchingProjects
+}
+
+// Only one rule has to be true for the project to match (ORed together).
+func nodeMatchesRules(node *NodeProjectData, rules []*iam_v2.ProjectRule) bool {
+	for _, rule := range rules {
+		if rule.Type == iam_v2.ProjectRuleTypes_NODE && nodeMatchesAllConditions(node, rule.Conditions) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// All the conditions must be true for a rule to be true (ANDed together).
+// If there are no conditions then the rule is false
+func nodeMatchesAllConditions(node *NodeProjectData, conditions []*iam_v2.Condition) bool {
+	if len(conditions) == 0 {
+		return false
+	}
+
+	for _, condition := range conditions {
+		switch condition.Attribute {
+		case iam_v2.ProjectRuleConditionAttributes_CHEF_SERVER:
+			values := node.getValues("chef_server")
+
+			if len(values) == 0 || !stringutils.SliceContains(condition.Values, values[0]) {
+				return false
+			}
+		case iam_v2.ProjectRuleConditionAttributes_CHEF_ORGANIZATION:
+			values := node.getValues("organization_name")
+
+			if len(values) == 0 || !stringutils.SliceContains(condition.Values, values[0]) {
+				return false
+			}
+		case iam_v2.ProjectRuleConditionAttributes_ENVIRONMENT:
+			values := node.getValues("environment")
+
+			if len(values) == 0 || !stringutils.SliceContains(condition.Values, values[0]) {
+				return false
+			}
+		case iam_v2.ProjectRuleConditionAttributes_CHEF_POLICY_GROUP:
+			values := node.getValues("policy_group")
+
+			if len(values) == 0 || !stringutils.SliceContains(condition.Values, values[0]) {
+				return false
+			}
+		case iam_v2.ProjectRuleConditionAttributes_CHEF_POLICY_NAME:
+			values := node.getValues("policy_name")
+
+			if len(values) == 0 || !stringutils.SliceContains(condition.Values, values[0]) {
+				return false
+			}
+		case iam_v2.ProjectRuleConditionAttributes_CHEF_ROLE:
+			foundMatch := false
+			values := node.getValues("roles")
+			if len(values) == 0 {
+				return false
+			}
+			for _, projectRole := range condition.Values {
+				if stringutils.SliceContains(values, projectRole) {
+					foundMatch = true
+					break
+				}
+			}
+			if !foundMatch {
+				return false
+			}
+		case iam_v2.ProjectRuleConditionAttributes_CHEF_TAG:
+			foundMatch := false
+			values := node.getValues("chef_tags")
+			if len(values) == 0 {
+				return false
+			}
+			for _, projectRole := range condition.Values {
+				if stringutils.SliceContains(values, projectRole) {
+					foundMatch = true
+					break
+				}
+			}
+			if !foundMatch {
+				return false
+			}
+		default:
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
There is probably a way to translate this to SQL but it'd probably be pretty messy. I think this being in golang is fine unless we run into performance problems or something.